ChemistryLib/PhreeqcKernelData/EquilibriumReactants.cpp

```diff
@@ -0,0 +1,37 @@
+/**
+ * \file
+ * \copyright
+ * Copyright (c) 2012-2019, OpenGeoSys Community (http://www.opengeosys.org)
+ *            Distributed under a Modified BSD License.
+ *              See accompanying file LICENSE.txt or
+ *              http://www.opengeosys.org/project/license
+ *
+ */
+
+#include "EquilibriumReactants.h"
+
+namespace ChemistryLib
+{
+namespace PhreeqcKernelData
+{
+PhaseComponent::PhaseComponent(std::string&& name,
+                               double const initial_amount,
+                               double const saturation_index)
+{
+    this->name = name;
+    moles = initial_amount;
+    si = saturation_index;
+    si_org = saturation_index;
+}
+
+EquilibriumReactants::EquilibriumReactants(
+    std::vector<PhaseComponent>& phase_components)
+{
+    for (auto& phase_component : phase_components)
+    {
+        auto name = phase_component.getName();
```

Comment:
```suggestion
        auto& name = phase_component.getName();
```
Why?

Reply:
Prevent copy, if my understanding is correct.